azurelinuxagent/common/osutil/nsbsd.py

```diff
@@ -153,3 +152,15 @@ def remove_firewall(self, dst_ip=None, uid=None):
     def enable_firewall(self, dst_ip=None, uid=None):
         # disable iptables methods
         return True
+
+    def supports_logrotate(self):
+        return False
+
+    def get_agent_log_dir(self):
+        return "/log/azure"
+
+    def get_agent_log_rotation_maxbytes(self):
+        return 4000000
+
+    def get_agent_log_rotation_backupcount(self):
+        return 1
```

Comment:
Its the configuration that @remip2 put it. I think it was because of the limited space available on the disks for `nsbsd` systems.

@remip2 Can you clarify this better?

Reply:
yes I confirm, my main concern for nsbsd is the disk space and in most cases only one backup is enough to get all the useful information from the log.

Reply:
@remip2 - Can you try the current fix on an NSBSD VM? (with the changes I did on top of yours) - Please let me know the findings.

Also, how can I create an NSBSD VM in Azure?

Reply:
ok I updated the agent on the existing vm and the rotation is working. I'll do a new  complete image and test the provisioning later.

To test you can deploy the vm from the marketplace : https://azuremarketplace.microsoft.com/fr-fr/marketplace/apps/stormshield.stormshield-elastic-virtual-appliance?tab=Overview
or via a github template : https://github.com/stormshield/azure-templates/tree/master/sns/sns-2-nics
The 'root' user is admin: ssh admin@<pub ip>
The files are located in: /usr/local/lib/python2.7/azurelinuxagent
and the log: /log/azure
To stop the agent : nstop waagent
To start the agent : nstart waagent

Reply:
I tested the provisioning from a new image, all is working as expected.

Reply:
Thanks @remip2 for verifying it - I verified it on an Ubuntu VM as well to see its rotation in action (The `cat`s in the middle were to inflate the size of the log file)

![image](https://user-images.githubusercontent.com/1047040/78619018-f147cf80-7830-11ea-9cc0-312e3033b001.png)

The only observation was that the need to archive the backup files. I'll see if I can send an update for it.

Reply:
@remip2  - Added support for archiving backup files - https://github.com/Azure/WALinuxAgent/pull/1714/commits/5bda83b4bc26153a12db2da013021ee4b1d6253e. Can you please let me know the default desired for NSBSD?

Reply:
@vrdmr I don't need gz for NSBSD, thanks